plugins/woocommerce/templates/single-product/product-image.php

```diff
@@ -24,19 +24,29 @@
 
 global $product;
 
+/**
+ * Controls the number of thumbnail per row displayed by flexslider.
+ *
+ * @since 2.7.0
+ *
+ * @param int $count The number of thumbnails displayed.
+ */
 $columns           = apply_filters( 'woocommerce_product_thumbnails_columns', 4 );
```

Comment:
Interested in your thoughts here (I realize the hook itself is pre-existing): if I add a filter setting this to anything but 4, then we lose the padding/margins around the thumbnails. Example with 5 columns (showing the absence of spacing between thumbnails):

<table>
<tr>
<td>
<img width="626" alt="4-expected" src="https://user-images.githubusercontent.com/3594411/174397757-153e6574-2fa6-467a-ad96-b10abf7773a6.png">
</td>
<td>
<img width="625" alt="5-no-padd-marg" src="https://user-images.githubusercontent.com/3594411/174397766-27bde2b9-ddca-4c89-bbbf-81d32847e65d.png">
</td>
</tr>
</table>

The other aspect is, if I set the number of columns to 3, the thumbnails increase in size to fill the space (this doesn't happen with `trunk`) and we end up with a rather blurry/pixellated effect; do you see the same?

Reply:
🤔 This may actually relate to [this code](https://github.com/woocommerce/woocommerce/pull/33041/files#diff-d49488c34017d273db858b7b6eff58f22d95310991d26638e75793c6a1e98678R1590)?

Reply:
the reason was the style of twentytwentyone.. has a custom forced style for 4 columns gallery. Is one of those things that I didn't want to change on my own, because will change the actual gallery appearance for the gallery. But good point I guess it needs to be fixed if you noticed it!
![image](https://user-images.githubusercontent.com/8550908/174402228-a1c9f74e-2aa8-4a98-9130-3ef4a17eee44.png)